src/crystal/main.cr

```diff
@@ -56,11 +56,15 @@ module Crystal
         1
       end
 
-    AtExitHandlers.run status
-    ex.inspect_with_backtrace STDERR if ex
+    # This will run after all at_exit handlers, to make sure the user sees the exception.
+    AtExitHandlers.do_last ->(code : Int32) do
+      exception = ex
+      exception.inspect_with_backtrace STDERR if exception
```

Comment:
`ex.try(&.inspect_with_backtrace(STDERR)`?

Reply:
Yes! that's a good idea, thank you!